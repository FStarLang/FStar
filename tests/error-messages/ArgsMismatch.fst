module ArgsMismatch

(* Attributes on arguments don't matter at all for applications. *)

let foo (#x:int) (y:int) = x+y

[@@expect_failure] let _ = foo 1 2
[@@expect_failure] let _ = foo #1 #2
                   let _ = foo #1 2
[@@expect_failure] let _ = foo 1 #2

let bar (#[@@@1]x:int) (y:int) = x+y

[@@expect_failure] let _ = bar 1 2
[@@expect_failure] let _ = bar #1 #2
                   let _ = bar #1 2
[@@expect_failure] let _ = bar 1 #2

let baz (#x:int) ([@@@2]y:int) = x+y

[@@expect_failure] let _ = baz 1 2
[@@expect_failure] let _ = baz #1 #2
                   let _ = baz #1 2
[@@expect_failure] let _ = baz 1 #2

let qux (#[@@@1]x:int) ([@@@2]y:int) = x+y

[@@expect_failure] let _ = qux 1 2
[@@expect_failure] let _ = qux #1 #2
                   let _ = qux #1 2
[@@expect_failure] let _ = qux 1 #2
