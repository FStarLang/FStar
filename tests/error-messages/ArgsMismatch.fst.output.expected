>> Got issues: [
* Error 173 at ArgsMismatch.fst(7,33-7,34):
  - Too many arguments to function of type y: Prims.int -> Prims.int
  - Got 2 arguments
  - Remaining type is Prims.int

>>]
>> Got issues: [
* Error 92 at ArgsMismatch.fst(8,35-8,36):
  - Inconsistent argument qualifiers.
  - Expected an implicit argument, got an explicit one.

>>]
>> Got issues: [
* Error 173 at ArgsMismatch.fst(10,34-10,35):
  - Too many arguments to function of type y: Prims.int -> Prims.int
  - Got 2 arguments
  - Remaining type is Prims.int

>>]
>> Got issues: [
* Error 173 at ArgsMismatch.fst(14,33-14,34):
  - Too many arguments to function of type y: Prims.int -> Prims.int
  - Got 2 arguments
  - Remaining type is Prims.int

>>]
>> Got issues: [
* Error 92 at ArgsMismatch.fst(15,35-15,36):
  - Inconsistent argument qualifiers.
  - Expected an implicit argument, got an explicit one.

>>]
>> Got issues: [
* Error 173 at ArgsMismatch.fst(17,34-17,35):
  - Too many arguments to function of type y: Prims.int -> Prims.int
  - Got 2 arguments
  - Remaining type is Prims.int

>>]
>> Got issues: [
* Error 173 at ArgsMismatch.fst(21,33-21,34):
  - Too many arguments to function of type [@@@ 2]y: Prims.int -> Prims.int
  - Got 2 arguments
  - Remaining type is Prims.int

>>]
>> Got issues: [
* Error 92 at ArgsMismatch.fst(22,35-22,36):
  - Inconsistent argument qualifiers.
  - Expected an implicit argument, got an explicit one.

>>]
>> Got issues: [
* Error 173 at ArgsMismatch.fst(24,34-24,35):
  - Too many arguments to function of type [@@@ 2]y: Prims.int -> Prims.int
  - Got 2 arguments
  - Remaining type is Prims.int

>>]
>> Got issues: [
* Error 173 at ArgsMismatch.fst(28,33-28,34):
  - Too many arguments to function of type [@@@ 2]y: Prims.int -> Prims.int
  - Got 2 arguments
  - Remaining type is Prims.int

>>]
>> Got issues: [
* Error 92 at ArgsMismatch.fst(29,35-29,36):
  - Inconsistent argument qualifiers.
  - Expected an implicit argument, got an explicit one.

>>]
>> Got issues: [
* Error 173 at ArgsMismatch.fst(31,34-31,35):
  - Too many arguments to function of type [@@@ 2]y: Prims.int -> Prims.int
  - Got 2 arguments
  - Remaining type is Prims.int

>>]
